security/fma-operator-fee.md

```diff
@@ -0,0 +1,118 @@
+  - [FM2: Broken Fee Estimation (Wallets)](#fm2-broken-fee-estimation-wallets)
+  - [FM3: Bug in Receipt Hydrating Logic](#fm3-bug-in-receipt-hydrating-logic)
+  - [FM4: Database Growth Impact on Nodes](#fm4-database-growth-impact-on-nodes)
+- [Action Items](#action-items)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+|                    |                                                    |
+| ------------------ | -------------------------------------------------- |
+| Author             | leruaa                                             |
+| Created at         | 2025-01-08                                         |
+| Initial Reviewers  |  Mark Tyneway                                      |
+| Need Approval From | Blaine Malone                                      |
+| Status             | In Review                                          |
+
+## Introduction
+
+This document covers initial deployment of Operator Fee.
+
+The OperatorFee is a new transaction fee that allows new OP chain variants to account for their unique cost structure. For example, the existing fee structure isn't friendly to chains using alt-DA's, since the l1fee only accounts for Ethereum's blobGasFee instead of an alt-DA's fee.
+
+Also, For OP Stack variants that want to utilize ZK proofs, the cost of ZK proving a transaction is a significant resource that is not taken into consideration in the current fee structure.
+
+Below are references for this project:
+
+- [Design Doc](../protocol/operator-fee.md)
+- [Spec](https://github.com/ethereum-optimism/specs/pull/382)
+
+## Failure Modes and Recovery Paths
+
+### FM1: Operator Fee scalars are set to incorrect values
+
+- **Description:** 
+  If the operator fee scalars are incorrectly initialized or updated, there is a risk that the transcations fees will be too high. This could lead to a situation where the chain become unusable.
+- **Risk Assessment:**
+  High impact, low likelihood.
+  **Mitigations:**
+  Before setting or updating the operator fee params, the operator should carefully read the [corresponding specs](https://specs.optimism.io/protocol/isthmus/exec-engine.html#operator-fee) and simulate the impact of operator fee on the whole transaction cost.
+- **Detection:** 
+  By default, the operator fee parameters are set to 0 and the feature is disabled. There are [E2E tests](https://github.com/ethereum-optimism/optimism/blob/develop/op-e2e/system/fees/fees_test.go) that ensure there is no impact on the transaction cost when the operator fee is disabled.
+
+  On chains that enable operator fee, the operator should monitor the transaction cost and ensure that the operator fee is not too high.
+- **Recovery Path(s)**:
+  If the operator fee parameters are set to unreasonable values, the rollup operator should update the `operatorFeeScalar` and `operatorFeeConstant` to reasonable values as soon as possible.
+
+### FM2: Broken Fee Estimation (Wallets)
```

Comment:
Related to this, should we have instructions for chain consumers (i.e. dune, l2beat etc) to be able to know which fees to be looking at? i.e. is it as simple as, if it's alt-da, then look at operator-fee?

Reply:
Also,  if operator fees are set, we want to make sure the `baseFeeScalar` and the `blobBaseFeeScalar` isn't set right?

Reply:
chain consumers could just look at transaction receipts and see if `operatorFeeConstant` and `operatorFeeScalar` fields are not 0.

Operator fees and `baseFeeScalar`, `blobBaseFeeScalar` can be used together. operator fee use case may be to account for additional costs, for instance ZK proving costs.